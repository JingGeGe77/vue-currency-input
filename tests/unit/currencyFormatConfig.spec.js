import currencyFormatConfig from '../../src/utils/currencyFormatConfig'

// eslint-disable-next-line no-global-assign
Intl = require('Intl')

describe('currencyFormatConfig', () => {
  it('returns the correct config', () => {
    expect(currencyFormatConfig({ currency: 'EUR', locale: 'de' })).toMatchSnapshot()
    expect(currencyFormatConfig({ currency: 'USD', locale: 'en' })).toMatchSnapshot()
    expect(currencyFormatConfig({ currency: 'JPY', locale: 'fr' })).toMatchSnapshot()
  })
})
