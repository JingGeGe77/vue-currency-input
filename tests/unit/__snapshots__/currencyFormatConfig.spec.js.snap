// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencyFormatConfig returns the correct config 1`] = `
Object {
  "allowDecimal": true,
  "decimalLimit": 2,
  "decimalSymbol": ",",
  "prefix": "",
  "suffix": " €",
  "thousandsSeparatorSymbol": ".",
}
`;

exports[`currencyFormatConfig returns the correct config 2`] = `
Object {
  "allowDecimal": true,
  "decimalLimit": 2,
  "decimalSymbol": ".",
  "prefix": "$",
  "suffix": "",
  "thousandsSeparatorSymbol": ",",
}
`;

exports[`currencyFormatConfig returns the correct config 3`] = `
Object {
  "allowDecimal": false,
  "decimalLimit": 0,
  "decimalSymbol": null,
  "prefix": "",
  "suffix": " JPY",
  "thousandsSeparatorSymbol": " ",
}
`;
